c2cgeoportal/views/checker.py

```diff
@@ -51,17 +50,20 @@ def build_url(name, url, request, headers=None):
         headers = {}
     headers["Cache-Control"] = "no-cache"
 
-    urlfragments = urlsplit(url)
-    if urlfragments.netloc == request.environ.get("SERVER_NAME") or \
-            urlfragments.netloc.startswith("localhost:"):
-        url_ = urlunsplit((
-            "http", "localhost", urlfragments.path, urlfragments.query, urlfragments.fragment
-        ))
-        headers["Host"] = urlfragments.netloc
-    else:
-        url_ = url
-
     settings = request.registry.settings.get("checker", {})
+    rewrite_as_http_localhost = settings.get("rewrite_as_http_localhost", True)
+    log.info("build_url rewrite_as_http_localhost: {}".format(rewrite_as_http_localhost))
+
+    url_ = url
+    if rewrite_as_http_localhost:
```

Comment:
Some sites do not allow access to http://localhost.
So the goal is to, with a configuration option, be able to override the hard-coded "redirection" of build_url to http://localhost.

Reply:
security policy of the site in question

Reply:
Witch kind of security policy, what does he check?

Reply:
http://localhost is not allowed on that server (Connection refused). Security policy is the one of the canton in question, we do not have control over it, and they will not open http://localhost. But if we simply keep the url instead of forcing it to use http://localhost, there is no problem.

Reply:
I don't like when we don't know more :-/, but so keep it ...